core/src/utils/transition/ios.transition.ts

```diff
@@ -123,56 +145,131 @@ const animateBackButton = (
   rootAnimation: Animation,
   rtl: boolean,
   backDirection: boolean,
-  backButtonEl: any,
-  largeTitleBox: DOMRect,
-  backButtonBox: DOMRect
+  backButtonEl: HTMLIonBackButtonElement,
+  backButtonBox: DOMRect,
+  backButtonTextEl: HTMLElement,
+  backButtonTextBox: DOMRect,
+  largeTitleEl: HTMLIonTitleElement,
+  largeTitleTextBox: DOMRect
 ) => {
   const BACK_BUTTON_START_OFFSET = rtl ? `calc(100% - ${backButtonBox.right + 4}px)` : `${backButtonBox.left - 4}px`;
-  const START_TEXT_TRANSLATE = rtl ? '7px' : '-7px';
-  const END_TEXT_TRANSLATE = rtl ? '-4px' : '4px';
-
-  const ICON_TRANSLATE = rtl ? '-4px' : '4px';
 
   const TEXT_ORIGIN_X = rtl ? 'right' : 'left';
   const ICON_ORIGIN_X = rtl ? 'left' : 'right';
 
+  const CONTAINER_ORIGIN_X = rtl ? 'right' : 'left';
+
+  /**
+   * When the title and back button texts match
+   * then they should overlap during the page transition.
+   * If the texts do not match up then the back button text scale adjusts
+   * to not perfectly match the large title text otherwise the
+   * proportions will be incorrect.
+   */
+  const doTitleAndButtonTextsMatch = backButtonTextEl.textContent === largeTitleEl.textContent;
+
+  const WIDTH_SCALE = largeTitleTextBox.width / backButtonTextBox.width;
+
+  /**
+   * We subtract 10px to account for slight sizing/padding
```

Comment:
```suggestion
   * We subtract an offset to account for slight sizing/padding
```
Optional but I can see this getting out of sync since the value is set so far away from the comment.